### PR TITLE
saml2aws: 2.15.0 -> 2.19.0

### PR DESCRIPTION
**Pull request by jpotier**
Current version is broken, at least for use with Okta and AWS. New
version fixes this problem.

Packaging has changed a bit too, deps.nix was removed, and the top-level
`buildGoModule` is used insead, since the upstream project now exposes a
go.mod file.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @pmyjavec

### CODE REVIEW

**Comment by jpotier** [comment]
maybe @teto for a review (saw you reviewed the last update)

File: pkgs/tools/security/saml2aws/default.nix [hunk 1]
```diff
@@ -1,24 +1,25 @@
-{ stdenv, buildGoPackage, fetchFromGitHub }:
+{ stdenv, buildGoModule, fetchFromGitHub }:
 
-buildGoPackage rec {
+buildGoModule rec {
   name = "saml2aws-${version}";
```
Comment (kalbasit) [inline pkgs/tools/security/saml2aws/default.nix]:
```suggestion
```

`pname`, `version` are already set.

File: pkgs/tools/security/saml2aws/default.nix [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
+
+  subPackages = [ "." "cmd/saml2aws" ];
+
+  buildFlagsArray = ''
+    -ldflags=-X main.Version=${version}
+  '';
+
   meta = with stdenv.lib; {
     description = "CLI tool which enables you to login and retrieve AWS temporary credentials using a SAML IDP";
     homepage    = https://github.com/Versent/saml2aws;
```
Comment (kalbasit) [inline pkgs/tools/security/saml2aws/default.nix]:
```suggestion
    homepage    = "https://github.com/Versent/saml2aws";
```

**Review by kalbasit — CHANGES_REQUESTED** [review]

File: pkgs/tools/security/saml2aws/default.nix (hunk 1 above)
Comment (jpotier) [reply pkgs/tools/security/saml2aws/default.nix]: noted

File: pkgs/tools/security/saml2aws/default.nix (hunk 2 above)
Comment (jpotier) [reply pkgs/tools/security/saml2aws/default.nix]: changed

**Comment by jpotier** [comment]
@kalbasit anything else to fix in this MR?

**Comment by teto** [comment]
nixpkgs-reviewd ok, the binary launches as well.

**Pull request merged**